.github/workflows/ci.yaml

```diff
@@ -82,7 +80,6 @@ jobs:
       - name: Checkout branch
         uses: actions/checkout@v3
         with:
-          ref: ${{ github.head_ref }}
           fetch-depth: 0
```

Comment:
nit: This fetches the entire history. Is this to run some backwards compatibility test?

Reply:
You mean it should be `fetch-depth: 1`? Which most likely is the default I guess.

I don't think there is any good reason why we do this.

Reply:
Yeah might be better to just remove the entire with block for simplicity.

Reply:
I think `fetch-depth` is used in our repositories because `setuptools-scm` uses Git tags to determine the version when building.